Networking/Networking/Remote/Remote.swift

```diff
@@ -22,34 +22,32 @@ public class Remote: NSObject {
     }
 
 
-    /// Enqueues the specified Network Request.
-    ///
-    /// - Parameters:
-    ///     - request: Request that should be performed.
-    ///     - completion: Closure to be executed upon completion. Will receive the JSON Parsed Response (if successful)
+    /// Enqueues the specified Network Request with a generic expected result type.
     ///
-    func enqueue(_ request: URLRequestConvertible, completion: @escaping (Any?, Error?) -> Void) {
-        network.responseData(for: request) { [weak self] (data, networError) in
-            guard let self = self else {
-                return
-            }
+    /// - Parameter request: Request that should be performed.
+    /// - Returns: The result from the JSON parsed response for the expected type.
+    func enqueue<T: Decodable>(_ request: URLRequestConvertible) async -> Result<T, Error> {
```

Comment:
Super nit: since we're using `async`, do you think we should also switch to using `throws` instead of `Result`? It's more common to see the `async throws` combination - I wonder if using that modernizes the code more.

Reply:
I used `Result` in the first place because the other most used `enqueue` with a mapper also returns a `Result`. Let's give `async throws` a try, updated in https://github.com/woocommerce/woocommerce-ios/pull/7897/commits/a2fc7526260c25e43772c580054beea78055cf01